pkg/apis/elasticsearch/v1/elasticsearch_config.go

```diff
@@ -38,6 +73,7 @@ type ElasticsearchSettings struct {
 
 // DefaultCfg is an instance of ElasticsearchSettings with defaults set as they are in Elasticsearch.
 var DefaultCfg = ElasticsearchSettings{
+	// Values below only make senses if there is no "node.roles" in the configuration provided by the user
```

Comment:
What happens if the user specifies `node.roles`? Do we still append our own `node.data`, `node.master`, etc. defaults in the Elasticsearch config?

Reply:
I don't think so because we apply the defaults only when reading the current Elasticsearch config but not when writing it.